packages/evm/src/opcodes/functions.ts

```diff
@@ -766,11 +766,24 @@ export const handlers: Map<number, OpHandler> = new Map([
   ],
   // 0x5b: JUMPDEST
   [0x5b, function () {}],
-  // 0x5c: BEGINSUB
+  // 0x5c: BEGINSUB or EXTSLOAD
   [
     0x5c,
-    function (runState) {
-      trap(ERROR.INVALID_BEGINSUB + ' at ' + describeLocation(runState))
+    function (runState, common) {
+      if (common.isActivatedEIP(2315)) {
+        // BEGINSUB is sync
+        trap(ERROR.INVALID_BEGINSUB + ' at ' + describeLocation(runState))
+      } else if (common.isActivatedEIP(2330)) {
```

Comment:
Hmm I'm not sure maybe we can do this in another way (no if/else branch), like inject them in `codes.ts`, the same way as how one would inject custom opcodes..?

Reply:
I agree the if-else doesn't look good here. I will explore moving logic in `codes.ts`. If you have an example can you please share?

Reply:
I think we might want to leverage this logic of "clashing opcodes" into `codes.ts` (we might also have this at some point with precompiles). The API is not (directly) ready for this, but https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/evm/tests/customOpcodes.spec.ts these tests might point a bit in the right direction.